sky/server/requests/executor.py

```diff
@@ -171,8 +172,14 @@ def process_request(self, executor: process.BurstableExecutor,
             # multiple requests can share the same process pid, which may cause
             # issues with SkyPilot core functions if they rely on the exit of
             # the process, such as subprocess_daemon.py.
-            executor.submit_until_success(_request_execution_wrapper,
-                                          request_id, ignore_return_value)
+            fut = executor.submit_until_success(_request_execution_wrapper,
+                                                request_id, ignore_return_value)
+            if retryable:
+                # If the task might fail and be retried, start a thread to
+                # monitor the future and process retry.
+                threading.Thread(target=self.handle_task_result,
+                                 args=(fut, request_element),
+                                 daemon=True).start()
 
             logger.info(f'[{self}] Submitted request: {request_id}')
         except (Exception, SystemExit) as e:  # pylint: disable=broad-except
```

Comment:
Will this error handling still apply to the retryable tasks?

Reply:
Ah, I see. I was confused by `submit_until_success` changing to return a future. But it was already asynchronous, and the submission process is still synchronous inside this try/except. Sounds good.